cpp/FileTransferAgent.hpp

```diff
@@ -59,6 +59,16 @@ class FileTransferAgent : public IFileTransferAgent
     m_uploadStreamSize = dataSize;
   }
 
+  /**
+   * Set useUrand to true to use /dev/urandom device
+   * Set it to false to use /dev/random device
+   * @param useUrand
+   */
+  virtual void setRandomDeviceAsUrand(bool useUrand)
+  {
+    m_useDevUrand = useUrand;
+  }
+
```

Comment:
This setRandomDeviceAsUrand() is invoked by filetransferagent and is used from ODBC and applications directly. 

The other setRandomDev is internal func in cpp/FileMetadataInitializer.hpp which sets the real random device to be used and the Encryptor uses the device value from this func.